sdk/storage/Azure.Storage.Files.Shares/tests/FileClientTests.cs

```diff
@@ -4690,6 +5206,51 @@ public async Task RenameAsync_SourceSasCredentialDestSasUri()
             Response<ShareFileProperties> response = await destDirectory.GetPropertiesAsync();
         }
 
+        [RecordedTest]
+        [TestCase(null)]
+        [TestCase(false)]
+        [TestCase(true)]
+        [ServiceVersion(Min = ShareClientOptions.ServiceVersion.V2022_11_02)]
+        public async Task RenameAsync_TrailingDot(bool? sourceAllowTrailingDot)
```

Comment:
TODO the Rename API inverts source and destination from the REST API.  Make sure x-ms-allow-trailing-dot and x-ms-source-allow-trailing-dot are wired up correctly.